Doc strings no longer generated for Python Enums

### What happened?

There appears to be a regression between 3.79.0 and 3.81.0 somewhere in the support for Python SDK generation. The problem is that with newer versions of Pulumi, documentation strings are no longer generated for Enum types. For example this is lost:

```
<     """
<     A flag to determine whether or not ChannelBinding is enabled or disabled.
<     """
```

### Expected Behavior

Python docstrings continue to be generated as before the change for enum types.

### Steps to reproduce

Makefile:

```
export PULUMI_IGNORE_AMBIENT_PLUGINS = true

repro::
	make v1
	rm -rf sdk/python-baseline
	mv sdk/python sdk/python-baseline
	make v2
	diff sdk/python-baseline/*/_enums.py sdk/python/*/_enums.py

v1::	.make/generate_python_1
v2::	.make/generate_python_2

.make/generate_python_1: .v1/.pulumi/bin/pulumi
	mkdir -p sdk/python
	.v1/.pulumi/bin/pulumi package gen-sdk schema.json --language python

.make/generate_python_2: .v2/.pulumi/bin/pulumi
	mkdir -p sdk/python
	.v2/.pulumi/bin/pulumi package gen-sdk schema.json --language python

.v1/.pulumi/bin/pulumi: HOME := ./.v1
.v1/.pulumi/bin/pulumi:
	curl -fsSL https://get.pulumi.com | sh -s -- --version "3.79.0"

.v2/.pulumi/bin/pulumi: HOME := ./.v2
.v2/.pulumi/bin/pulumi:
	curl -fsSL https://get.pulumi.com | sh -s -- --version "3.81.0"
```

Now schema.json:

```
{
  "name": "azure-native",
  "description": "A Pulumi package.",
  "keywords": [
    "pulumi"
  ],
  "homepage": "https://pulumi.io",
  "license": "Apache-2.0",
  "attribution": "TBD",
  "repository": "https://github.com/pulumi/pulumi-example",
  "meta": {
    "moduleFormat": "(.*)(?:/[^/]*)"
  },
  "language": {
    "python": {
      "moduleNameOverrides": {
        "aad": "aad"
      },
      "readme": "The native Azure provider package offers...",
      "requires": {
        "pulumi": ">=3.35.0,<4.0.0"
      },
      "usesIOClasses": true
    }
  },
  "config": {},
  "resources": {
    "azure-native:aad:ExampleResource": {
        "inputProperies": {
            "p": {
                "type": {
                    "$ref": "!#/types/azure-native:aad:ChannelBinding"
                }
            }
        }
    }
  },
  "types": {
    "azure-native:aad:ChannelBinding": {
      "description": "A flag to determine whether or not ChannelBinding is enabled or disabled.",
      "type": "string",
      "enum": [
        {
          "value": "Enabled"
        },
        {
          "value": "Disabled"
        }
      ]
    }
  }
}
```

You can observe `make repro`:

```
make repro                                                                                                                                                                                                                                                                                                ~/tmp/2023/09/repro-aznative-2736
make v1
mkdir -p sdk/python
.v1/.pulumi/bin/pulumi package gen-sdk schema.json --language python
rm -rf sdk/python-baseline
mv sdk/python sdk/python-baseline
make v2
mkdir -p sdk/python
.v2/.pulumi/bin/pulumi package gen-sdk schema.json --language python
diff sdk/python-baseline/*/_enums.py sdk/python/*/_enums.py
2c2
< # *** WARNING: this file was generated by pulumi. ***
---
> # *** WARNING: this file was generated by pulumi-language-python. ***
13,15d12
<     """
<     A flag to determine whether or not ChannelBinding is enabled or disabled.
<     """
make: *** [repro] Error 1
[2] t0yv0@Antons-MacBook-Pro>                                                  
```

### Output of `pulumi about`

The regression is between 3.79.0 and 3.81.0 somewhere.

### Additional context

This is blocking azure-native routine update to 3.81.0.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/changelog/pending/20230912--sdkgen-go-nodejs-python--fix-a-bug-in-marshalling-enums-across-grpc.yaml
+++ b/changelog/pending/20230912--sdkgen-go-nodejs-python--fix-a-bug-in-marshalling-enums-across-grpc.yaml
@@ -1,0 +1,4 @@
+changes:
+- type: fix
+  scope: sdkgen/go,nodejs,python
+  description: Fix a bug in marshalling enums across gRPC

--- a/pkg/codegen/schema/schema.go
+++ b/pkg/codegen/schema/schema.go
@@ -1083,8 +1083,12 @@ func (pkg *Package) marshalEnum(t *EnumType) ComplexTypeSpec {
 	}
 
 	return ComplexTypeSpec{
-		ObjectTypeSpec: ObjectTypeSpec{Type: pkg.marshalType(t.ElementType, false).Type, IsOverlay: t.IsOverlay},
-		Enum:           values,
+		ObjectTypeSpec: ObjectTypeSpec{
+			Description: t.Comment,
+			Type:        pkg.marshalType(t.ElementType, false).Type,
+			IsOverlay:   t.IsOverlay,
+		},
+		Enum: values,
 	}
 }
 

--- a/pkg/codegen/schema/schema_test.go
+++ b/pkg/codegen/schema/schema_test.go
@@ -96,6 +96,42 @@ func TestRoundtripLocalTypeRef(t *testing.T) {
 	assert.Empty(t, diags)
 }
 
+func TestRoundtripEnum(t *testing.T) {
+	// Regression test for https://github.com/pulumi/pulumi/issues/13921
+	t.Parallel()
+
+	assertEnum := func(t *testing.T, pkg *Package) {
+		typ, ok := pkg.GetType("enum:index:Color")
+		assert.True(t, ok)
+		enum, ok := typ.(*EnumType)
+		assert.True(t, ok)
+		assert.Equal(t, "An enum representing a color", enum.Comment)
+		assert.ElementsMatch(t, []*Enum{
+			{Value: "red"},
+			{Value: "green"},
+			{Value: "blue"},
+		}, enum.Elements)
+	}
+
+	testdataPath := filepath.Join("..", "testing", "test", "testdata")
+	loader := NewPluginLoader(utils.NewHost(testdataPath))
+	pkgSpec := readSchemaFile("enum-1.0.0.json")
+	pkg, diags, err := BindSpec(pkgSpec, loader)
+	require.NoError(t, err)
+	assert.Empty(t, diags)
+	assertEnum(t, pkg)
+
+	newSpec, err := pkg.MarshalSpec()
+	require.NoError(t, err)
+	require.NotNil(t, newSpec)
+
+	// Try and bind again
+	pkg, diags, err = BindSpec(*newSpec, loader)
+	require.NoError(t, err)
+	assert.Empty(t, diags)
+	assertEnum(t, pkg)
+}
+
 func TestImportSpec(t *testing.T) {
 	t.Parallel()
 

--- a/pkg/codegen/testing/test/testdata/enum-1.0.0.json
+++ b/pkg/codegen/testing/test/testdata/enum-1.0.0.json
@@ -1,0 +1,25 @@
+{
+    "$schema": "https://raw.githubusercontent.com/pulumi/pulumi/master/pkg/codegen/schema/pulumi.json",
+    "name": "enum",
+    "version": "1.0.0",
+    "//": [
+        "Simple schema with an enum"
+    ],
+    "types": {
+        "enum:index:Color": {
+            "description": "An enum representing a color",
+            "type": "string",
+            "enum": [
+                {
+                    "value": "red"
+                },
+                {
+                    "value": "green"
+                },
+                {
+                    "value": "blue"
+                }
+            ]
+        }
+    }
+}

--- a/pkg/codegen/testing/utils/host.go
+++ b/pkg/codegen/testing/utils/host.go
@@ -89,5 +89,6 @@ func NewHost(schemaDirectoryPath string) plugin.Host {
 		SchemaProvider{"using-dashes", "1.0.0"},
 		SchemaProvider{"auto-deploy", "0.0.1"},
 		SchemaProvider{"localref", "1.0.0"},
+		SchemaProvider{"enum", "1.0.0"},
 	)
 }

